src/main/java/com/jcraft/jsch/Channel.java

```diff
@@ -128,10 +125,8 @@ static void del(Channel c){
   int notifyme=0; 
 
   Channel(){
-    synchronized(pool){
-      id=index++;
-      pool.addElement(this);
-    }
+    id=index.incrementAndGet();
```

Comment:
Shouldn't this be `id=index.getAndIncrement()` in order to maintain the same semantics as `id=index++`?